filebeat/tests/system/test_harvester.py

```diff
@@ -778,47 +778,3 @@ def test_truncate(self):
         # Check that only 1 registry entry as original was only truncated
         data = self.get_registry()
         assert len(data) == 1
-
-    def test_decode_error(self):
-        """
-        Tests that in case of a decoding error it is handled gracefully
-        """
-        self.render_config_template(
-            path=os.path.abspath(self.working_dir) + "/log/*",
-            encoding="utf-16be",
```

Comment:
I would keep these tests as they not only tests if the decoding happens correctly but also if the config options work which can't be covered in the unit tests.

Reply:
This test was actually testing a bug... The error message the test looks for should never have been printed in first place.

Reply:
not sure I fully understand this comment. And I think there are 2 things here:

* Testing the config option
* Testing the conversion. 

What I think we should keep is the first point.

Reply:
The test didn't do any of these correctly. We tripped on it a few times, wondering why/how it actually works. After fixing the bug, we just couldn't make sense of the test anymore.